### DISCUSSION TITLE
Allow passing other values for setEnabledForSite on propagating via console

### DISCUSSION THREAD

**Discussion by vnali** [craftcms/cms#13081] [General]
Since 4.4.7, it is possible to propagate elements via console resaveController
I am trying to propagate one element to another site as disabled
 - If I don't provide --setEnabledForSite, it seems on this line `$this->setEnabledForSite?? $siteStatuses[$siteId]?? $element->getEnabledForSite()`,  because the current status of `$siteStatuses[$siteId]` also is not set, `$element->getEnabledForSite()` is used and it seems it returns true always 
- If I specify --setEnabledForSite even like `--setEnabledForSite=false`, `$this->setEnabledForSite` returns true  so element is enabled anyway,

.......
**Update:**
It seems it is possible to set to disable if we pass like `--setEnabledForSite=0`

**although** I have an idea if Craft allows passing setEnabledForSite as `--setEnabledForSite=0,1,0 --propagate-to=en1,en2,en3`
and also entry default status for the site being used if --setEnabledForSite is not provided

**Comment by brandonkelly** [comment]
>     * If I don't provide --setEnabledForSite, it seems on this line `$this->setEnabledForSite?? $siteStatuses[$siteId]?? $element->getEnabledForSite()`,  because the current status of `$siteStatuses[$siteId]` also is not set, `$element->getEnabledForSite()` is used and it seems it returns true always

It’s expected that it would use the current site status as the default value, yes.

> ....... **Update:** It seems it is possible to set to disable if we pass like `--setEnabledForSite=0`

Correct, that’s how all boolean CLI options work in Yii/Craft. No value means `true`, and `=0` means `false`.

> **although** I have an idea if Craft allows passing setEnabledForSite as `--setEnabledForSite=0,1,0 --propagate-to=en1,en2,en3` and also entry default status for the site being used if --setEnabledForSite is not provided

That’s a nice idea, however we can’t make that possible without dropping the `bool` type declaration, which would break the ability to pass `--set-enabled-for-site` without a value.

You could work around that by executing the command twice though – one for all the sites where the entries should be propagated as enabled, and another time for the sites where the entries should be disabled.
